app/src/main/java/com/android/unio/ui/event/EventListOverview.kt

```diff
@@ -0,0 +1,382 @@
+  var followingTabXCoordinate by remember { mutableStateOf(0f) }
+
+  val horizontalHeaderPadding = 16.dp
+
+  Scaffold(
+      floatingActionButton = {
+        FloatingActionButton(onClick = onAddEvent, modifier = Modifier.testTag("event_MapButton")) {
+          Icon(imageVector = Icons.Filled.Add, contentDescription = "Add Event")
+        }
+      },
+      content = { paddingValues ->
+        Column(modifier = Modifier.fillMaxSize().padding(paddingValues).background(Color.Black)) {
+          // Sticky Header
+          Box(
+              modifier =
+                  Modifier.fillMaxWidth()
+                      .background(Color.Black)
+                      .padding(vertical = 16.dp, horizontal = horizontalHeaderPadding)
+                      .testTag("event_Header")) {
+                Column {
+                  Text(
+                      text = "Upcoming Events",
+                      fontWeight = FontWeight.Bold,
+                      color = Color.White,
+                      style =
+                          TextStyle(
+                              fontSize =
+                                  24.sp // Set the font size to 24 sp (scale-independent pixels)
+                              ))
+                  Spacer(modifier = Modifier.height(8.dp))
+
+                  // Row with clickable tabs
+                  Row(
+                      modifier = Modifier.fillMaxWidth(),
+                      horizontalArrangement = Arrangement.SpaceBetween // Aligns the tabs evenly
+                      ) {
+                        // Clickable text for "All"
+                        Text(
+                            text = "All",
+                            color = if (selectedTab == "All") Color.White else Color.Gray,
+                            modifier =
+                                Modifier.clickable {
+                                      selectedTab = "All"
+                                      coroutineScope.launch {
+                                        animatablePosition.animateTo(
+                                            0f, // Starting point for "All"
+                                            animationSpec =
+                                                tween(durationMillis = 1000) // Animation duration
+                                            )
+                                      }
+                                    }
+                                    .padding(horizontal = 16.dp)
+                                    .onGloballyPositioned { coordinates ->
+                                      allTabWidth = (coordinates.size.width / density).dp
+                                      allTabXCoordinate = coordinates.positionInRoot().x
+                                    }
+                                    .testTag("event_tabAll"))
+
+                        // Clickable text for "Following"
+                        Text(
+                            text = "Following",
```

Comment:
It's a good habit to develop to avoid writing hard coded string in the code, and favor the use of String.xml resources. This will makes it easier to later [localize](https://developer.android.com/guide/topics/resources/localization) the application. It also makes the app more maintainable as it is easy to modify one text, without having to remember to modify every places that are displaying the hard coded text.